### DISCUSSION TITLE
Update resolver syntax

### DISCUSSION THREAD

**Discussion by chrisschaub** [MichalLytek/type-graphql#1285] [Q&A]
I have been trying to get an update mutation to work correctly, and I can't get past an unkown type error:

In my resolver definition, I have
```
  @Mutation(returns => Customer)
  async updateCustomer(
    @Arg("updateCustomerInput") updateCustomerInput: UpdateCustomer,
    @Ctx() ctx: Context,
  ): Promise<Customer> {
    return ctx.prisma.customer.update({
      where: { id: updateCustomerInput.id },
      data: { 
        name: updateCustomerInput.name,
        email: updateCustomerInput.email,
        user_id: updateCustomerInput.user_id,
        updated: Date.now()
      },
    })
  }
```
I have the following input defined
```
@InputType()
export class UpdateCustomer implements Partial<Customer> {
  @Field(type => ID)
  readonly id: string

  @Field()
  name: string

  @Field()
  email: string

  @Field()
  user_id: string
}
```
I've keep getting this error:
```
react_devtools_backend.js:4026 Error: Unknown type "updateCustomerInput".: {"response":{"errors":[{"message":"Unknown type \"updateCustomerInput\".","extensions":{"code":"GRAPHQL_VALIDATION_FAILED"...
```
The query comes in as
```
{
  "query": "mutation ($input: updateCustomerInput) {\n  updateCustomer (input: $input) {\n    id, name, email, user_id, created, updated\n  }\n}",
  "variables": {
    "input": {
      "where": {
        "id": "6286b2a73ca0f733ecd4bc46"
      },
      "data": {
        "id": "6286b2a73ca0f733ecd4bc46",
        "name": "Chris Test",
        "email": "chris@schaub.com",
        "user_id": "123"
      }
    }
  }
}
```
I just am stuck on the correct styntax for defining this update mutation. My other queries are working, but have been looking at this, trying to find a better example, but there are not many update examples anywhere. Ok, thanks for any help!

**Comment by MichalLytek** [comment]
Show the document (mutation) you're sending in a readable format

**Reply by MichalLytek** [reply]
`updateCustomer (input: $input)`
it's wrong because you've set the name to `updateCustomerInput`

**Comment by chrisschaub** [comment]
Sorry, reformatted the json query that is being sent

**Reply by chrisschaub** [reply]
I've tried changing the name to "input". Can you point to an example of an update so I can compare and stare at it? I am not writing the query, just the backend. But the query looks pretty standard. Just looking for a guide to the syntax for an update mutation. Thanks again.

**Reply by MichalLytek** [reply]
You should have autocomplete and errors in the graphql playground / apollo studio.
